### ISSUE TITLE
Correctness of the list of FVTR tests being ignored on AT next

### ISSUE THREAD

**Issue by mscastanho** [advancetoolchain/advance-toolchain#1813]
This is the summary of FVTR tests from a recent AT next build:

```
amino			power8: 	ignored
boost			power8: 	passed
ck_binaries		power8: 	passed
ck_gold			power8: 	ignored
ck_ldconfig		power8: 	passed
ck_ldds			power8: 	passed
ck_libbits		        power8: 	passed
ck_lock_elision        power8: 	ignored
ck_provides		power8: 	passed
ck_requires		power8: 	passed
ck_tuned		        power8: 	ignored
dhrystone		power8: 	ignored
gcc-builtins		power8: 	passed
gccgo			power8: 	ignored
gdb			        power8: 	ignored
gmp			power8: 	passed
golang			power8: 	ignored
gomp			power8: 	passed
gotools			power8: 	passed
libauxv			power8: 	ignored
libdfp			power8: 	passed
libexpat		        power8: 	passed
libhugetlbfs		power8: 	passed
libnxz			power8: 	passed
libvecpf		        power8: 	passed
mpfr			power8: 	passed
openssl			power8: 	passed
oprofile		        power8: 	ignored
package-check	power8: 	passed
paflib			power8: 	passed
perf			        power8: 	ignored
ppc476-workaround	power8: 	ignored
pthreads		        power8: 	passed
pveclib			power8: 	passed
python			power8: 	ignored
speccpu2006		power8: 	ignored
speccpu2017		power8: 	ignored
sphde			power8: 	passed
sysroot			power8: 	passed
systemtap		power8: 	passed
tbb			        power8: 	passed
tcmalloc		        power8: 	passed
timezone		        power8: 	ignored
urcu			power8: 	passed
valgrind		        power8: 	passed
xlc			        power8: 	ignored
xlcdfp			power8: 	ignored
xlf			        power8: 	ignored
zlib			        power8: 	passed
**************** TEST SUMMARY *****************
Passed tests: 30
Skipped tests: 19
Failed tests: 0
***********************************************
```

Some of the tests I believe are being correctly ignored, but I'm not sure about others. These are the ones that caught my attention (some with the output):
- Python (fixed by #1812).
- ck_tuned
```
ck_tuned: power8:  Start logging...
ck_tuned: power8:  Tuned libraries had different settings prior to AT 7.0.
```
- perf
```
perf: power8:  Start logging...
perf: power8:  /sys/kernel/debug/tracing/uprobe_events is not writable.
perf: power8:    You may be able to resolve this by:
perf: power8:    $ /usr/bin/sudo mount -o remount,mode=755 /sys/kernel/debug
perf: power8:    $ /usr/bin/sudo mount -o remount,mode=755 /sys/kernel/debug/tracing
perf: power8:    $ /usr/bin/sudo chmod a+rw /sys/kernel/debug/tracing/uprobe_events
perf: power8:  Skipping...
```
- gdb (looks like the same problem as for Python...) (fixed by #1812)
```
gdb: power8:  Start logging...
gdb: power8:  This test doesn't work on AT 5.0 or 6.0.
gdb: power8:  Skipping...
```
- timezone: The logs show success, but the output shows `ignored`
```
timezone: power8:  Start logging...
timezone: power8:  Skipping dynamic linking: running test suite on a compatible mode with the build system.     [SUCCESS]
```

Also, if we don't distribute go anymore, why do we still run `gotools` test?

**Comment by tuliom** [comment]
Besides the ones you listed, the following tests should have not been ignored:
- ck_gold 
- ck_lock_elision
- ck_tuned
- gccgo

Feels like they will all need changes like #1812 .
Would it be possible to stop duplicating code and move this to a shared place?

> Also, if we don't distribute go anymore, why do we still run gotools test?

@mscastanho because gcc still provides go.

**Comment by mscastanho** [comment]
>  Would it be possible to stop duplicating code and move this to a shared place?
@tuliom Agreed. I'll leave the fixes from #1812 as-is since they are already being tested, and I can move that check to a shared place as part an upcoming PR.

**Comment by ThinkOpenly** [comment]
> * perf
> 
> ```
> perf: power8:  Start logging...
> perf: power8:  /sys/kernel/debug/tracing/uprobe_events is not writable.
> perf: power8:    You may be able to resolve this by:
> perf: power8:    $ /usr/bin/sudo mount -o remount,mode=755 /sys/kernel/debug
> perf: power8:    $ /usr/bin/sudo mount -o remount,mode=755 /sys/kernel/debug/tracing
> perf: power8:    $ /usr/bin/sudo chmod a+rw /sys/kernel/debug/tracing/uprobe_events
> perf: power8:  Skipping...
> ```

Running the perf tests would require the suggested modifications to the test system(s).

**Comment by rzinsly** [comment]
>     * timezone: The logs show success, but the output shows `ignored`
> 
> 
> ```
> timezone: power8:  Start logging...
> timezone: power8:  Skipping dynamic linking: running test suite on a compatible mode with the build system.     [SUCCESS]
> ```

This is the same behaviour of all tests that don't run in the compatible mode, like the XL tests.

**Comment by mscastanho** [comment]
I already have a PR to fix most issues. I'm just waiting for #1812 because they may conflict and I want rebase over that once it gets merged.

The only one that would still be missing is the perf test, which actually requires changes to the test servers.

**Comment by mscastanho** [comment]
All faulty regex checks were fixed by #1825 and now all build servers are properly configured to run the perf tests.
